pkg/apis/pipeline/v1alpha1/image_resource_test.go

```diff
@@ -0,0 +1,85 @@
+/*
+Copyright 2019 The Tekton Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1_test
+
+import (
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+
+	"github.com/tektoncd/pipeline/pkg/apis/pipeline/v1alpha1"
+	tb "github.com/tektoncd/pipeline/test/builder"
+)
+
+func Test_Invalid_NewImageResource(t *testing.T) {
```

Comment:
it looks like the naming format (use of underscores) lines up with `git_resource_test.go` and `build_gcs_resource_test.go` but not any of the other test files. It [doesn't seem idiomatic](https://golang.org/pkg/testing/) to put an underscore straight after the word Test in the func name. I suggest renaming this to something like `TestNewImageResource_Invalid`.  Same for the other functions below.  This will bring the names more in line with e.g. `taskrun_validation_test.go`(with func `TestTaskRun_Validate`), `condition_types_test.go`( with func `TestConditionCheck_IsSuccessful`), `dag_test.go`(with func `TestBuild_Parallel`), etc...  In all of these the underscore is used to separate only the specific variation in testing condition / outcome that's going to be exercised.